pkg/configs/api/api.go

```diff
@@ -111,7 +113,7 @@ func (a *API) setConfig(w http.ResponseWriter, r *http.Request) {
 	logger := util.WithContext(r.Context(), util.Logger)
 
 	var cfg configs.Config
-	if err := json.NewDecoder(r.Body).Decode(&cfg); err != nil {
+	if err := yaml.NewDecoder(r.Body).Decode(&cfg); err != nil {
```

Comment:
Instead of assuming JSON is a subset of YAML for this to work (works most of the time, but there are corner cases), can you use a content-type head to detect YAML?  And have JSON be the default.